src/ussd_popi_user_data.js

```diff
@@ -420,32 +420,112 @@ go.app = function() {
                     }
                 },
                 next: function(content) {
-                    // TODO: Update phone number on identitystore
-                    return 'state_updated';
+                    return 'state_check_msisdn_available';
+                }
+            });
+        });
+        
+        self.add('state_check_msisdn_available', function(name) {
+            var new_msisdn = utils.normalize_msisdn(self.im.user.answers.state_new_msisdn, '27');
+            self.im.user.set_answer("new_msisdn", new_msisdn);
+
+            return is
+            .list_by_address({msisdn: new_msisdn})
+            .then(function(identities_found) {
+                var identities = (identities_found.results.length > 0) ? identities_found.results : null;
+
+                if (identities === null) {
+                    return self.states.create('state_create_msisdn_change');
+                }
+
+                var opted_out_on_operator = false;
+                var msisdn_available = true;
+                for(i=0; i < identities.length; i++) {
```

Comment:
We can probably use lodash's forEach here, instead of the `for(i=...) {identities[i]...}` pattern here, to make things a little cleaner: https://lodash.com/docs/4.17.4#forEach